crates/hyperswitch_domain_models/src/merchant_account.rs

```diff
@@ -0,0 +1,287 @@
+use common_utils::{
+    crypto::{OptionalEncryptableName, OptionalEncryptableValue},
+    date_time,
+    errors::{CustomResult, ValidationError},
+    ext_traits::ValueExt,
+    pii,
+};
+use diesel_models::{
+    encryption::Encryption, enums::MerchantStorageScheme,
+    merchant_account::MerchantAccountUpdateInternal,
+};
+use error_stack::ResultExt;
+use masking::{PeekInterface, Secret};
+use router_env::logger;
+
+use crate::type_encryption::{decrypt, AsyncLift};
+
+#[derive(Clone, Debug, serde::Serialize)]
+pub struct MerchantAccount {
```

Comment:
Aren't we handling V2 MerchantAccount storage object?

Reply:
No, the storage object is the same currently. Will need to find a way to have separate storage objects based on the feature flag.